components/ai_chat/browser/ai_chat_tab_helper.cc

```diff
@@ -336,7 +336,8 @@ void AIChatTabHelper::MaybeGeneratePageText() {
     FetchPageContent(
         web_contents(),
         base::BindOnce(&AIChatTabHelper::OnTabContentRetrieved,
-                       weak_ptr_factory_.GetWeakPtr(), current_navigation_id_));
+                       weak_ptr_factory_.GetWeakPtr(), current_navigation_id_),
+        HasUserOptedIn());
```

Comment:
If `HasUserOptedIn()` is `false` then `FetchPageContent` doesn't do anything (if it has to make a fetch). So in those cases it won't get content. So why not avoid that complication by not calling `FetchPageContent`? As discussed, the opt-in UX doesn't need to know anything about the page content, it only needs to know if the page has content. At the moment all pages have content if they are http(s). Can't we just send that data to the page's opt-in UX so it can differ whether it shows the page content association or not? For example, `SiteInfo` can simply get an `{ isPresent: boolean }` new property as well as `title` becoming optional.

Reply:
If the user hasn't opted in, FetchPageContent will still be called. Previously, we would gate the call with various conditions, including the user not having opted in. However, now we send that boolean through. We gate it prior to sending results, and in the case of video, we don't make a network request.